tests/functional/event/test_events_emission.py

```diff
@@ -0,0 +1,176 @@
+
+    def _bt_make_connections(self, bt_list):
+        for bt_entry in bt_list:
+            self.bt_connections.append(
+                Beanstalk.from_url('beanstalk://{0}'.format(bt_entry['addr'])))
+
+    def _bt_watch(self, tube):
+        for bt in self.bt_connections:
+            bt.watch(tube)
+
+    def _bt_pull_events_by_type(self, type):
+        pulled_events = []
+
+        for bt in self.bt_connections:
+            job_id = True
+            while job_id is not None:
+                try:
+                    job_id, data_raw = bt.reserve(timeout=4)
+                    pulled_events.append(json.loads(data_raw))
+                    bt.delete(job_id)
+                except ResponseError:
+                    break
+
+        return list(
+            filter(lambda x: x.get("event") == type,
+                   pulled_events))
+
+    def _service(self, name, action):
+        name = "%s-%s" % (self.conf['namespace'], name)
+        check_call(['gridinit_cmd', '-S',
+                    os.path.expanduser('~/.oio/sds/run/gridinit.sock'),
+                    action, name])
+        """
+        The current gridinit version that's running on travis returns before
+        the child process terminates. As such we have to introduce an
+        artificial pause to account for the child process termination. Remove
+        this once gridinit is upgraded.
+        """
+        time.sleep(3)
+
+    def test_container_create(self):
+        if len(self.bt_connections) > 1:
+            self.skipTest("Unsupported on multi-beanstalk setups.")
+
+        # First shutdown the event-agent
+        self._service(self.event_agent_name, 'stop')
+        self._bt_watch(DEFAULT_TUBE)
+
+        # Fire up the event
+        self.container_client.container_create(self.account,
+                                               self.container_name)
+
+        # Grab all events and filter for the needed event type
+        wanted_events = self._bt_pull_events_by_type(
+            EventTypes.ACCOUNT_SERVICES)
+
+        self.assertEqual(len(wanted_events), 1)
+        # Prepping for the next operation.
+        ev = wanted_events[0]
+
+        # Basic info
+        self.assertEqual(ev.get("url"), {
+            'ns': self.namespace,
+            'account': self.account,
+            'user': self.container_name,
+            'id': self.container_id,
+            'type': 'meta2'
+        })
+
+        # Get the peers list and verify it's the same as received
+        raw_dir_info = self.storage_api.directory.list(self.account,
+                                                       self.container_name,
+                                                       cid=self.container_id)
+        raw_dir_info = raw_dir_info['srv']
+        expected_peers_list = list(
+            filter(lambda x: x.get('type') == 'meta2', raw_dir_info))
+        expected_peers_list = list(
+            map(lambda x: x.get('host'), expected_peers_list))
```

Comment:
`filter` and `map` have been deprecated in favor of list comprehensions.
See https://docs.quantifiedcode.com/python-anti-patterns/readability/using_map_or_filter_where_list_comprehension_is_possible.html

Reply:
Fixed in adf2396